SpatialGDK/Source/SpatialGDK/Private/Utils/ActorGroupManager.cpp

```diff
@@ -0,0 +1,202 @@
+			ClassPathToActorGroup.Add(TSoftClassPtr<AActor>(Class), ActorGroup);
+			return ActorGroup;
+		}
+	}
+	return NAME_None;
+}
+
+FName UActorGroupManager::GetWorkerTypeForClass(UClass* Class)
+{
+	FName ActorGroup = GetActorGroupForClass(Class);
+
+	if (ActorGroupToWorkerType.Contains(ActorGroup))
+	{
+		return ActorGroupToWorkerType.FindRef(ActorGroup);
+	}
+
+	return NAME_None;
+}
+
+#if WITH_EDITOR
+
+bool GetFirstDifferentValue(TArray<FName> A, TArray<FName> B, FName& OutDifferent)
+{
+	for (const FName AName : A)
+	{
+		if (!B.Contains(AName))
+		{
+			OutDifferent = AName;
+			return true;
+		}
+	}
+	return false;
+}
+
+void UActorGroupManager::ValidateOffloadingSettings(TMap<FName, FActorClassSet> OldActorGroups, TMap<FName, FActorClassSet>* ActorGroups,
+	TSet<FName> OldWorkerTypes, TSet<FName>* WorkerTypes, FWorkerAssociation& WorkerAssociation)
+{
+	if (ActorGroups->Num() == 0)
+	{
+		ActorGroups->Append(UActorGroupManager::DefaultActorGroups());
+	}
+
+	// Check for empty Worker Types
+	if (WorkerTypes->Num() == 0)
+	{
+		WorkerTypes->Append(UActorGroupManager::DefaultWorkerTypes());
+	}
+
+	TArray<FName> ActorGroupKeys;
+	ActorGroups->GetKeys(ActorGroupKeys);
+
+	TArray<FName> OldActorGroupKeys;
+	OldActorGroups.GetKeys(OldActorGroupKeys);
+
+	// Check for renamed Actor Group
+	if (ActorGroups->Num() == OldActorGroups.Num())
+	{
+		FName FromActorGroup, ToActorGroup;
+		if (GetFirstDifferentValue(OldActorGroupKeys, ActorGroupKeys, FromActorGroup)
+			&& GetFirstDifferentValue(ActorGroupKeys, OldActorGroupKeys, ToActorGroup))
+		{
+			if (WorkerAssociation.ActorGroupToWorker.Contains(FromActorGroup))
+			{
+				WorkerAssociation.ActorGroupToWorker.Add(ToActorGroup, WorkerAssociation.ActorGroupToWorker.FindRef(FromActorGroup));
+				WorkerAssociation.ActorGroupToWorker.Remove(FromActorGroup);
+			}
+		}
+	}
+
+	// Check for invalid WorkerType (Currently just UnrealClient)
+	if (WorkerTypes->Contains(FName(*SpatialConstants::ClientWorkerType)))
+	{
+		WorkerTypes->Remove(FName(*SpatialConstants::ClientWorkerType));
+	}
+
+	// Check for renamed WorkerType
+	if (WorkerTypes->Num() == OldWorkerTypes.Num())
+	{
+		FName FromWorkerType, ToWorkerType;
+		if (GetFirstDifferentValue(OldWorkerTypes.Array(), WorkerTypes->Array(), FromWorkerType) &&
+			GetFirstDifferentValue(WorkerTypes->Array(), OldWorkerTypes.Array(), ToWorkerType))
+		{
+			for (auto Entry = WorkerAssociation.ActorGroupToWorker.CreateConstIterator(); Entry; ++Entry)
+			{
+				if (Entry->Value == FromWorkerType)
+				{
+					WorkerAssociation.ActorGroupToWorker.Add(Entry->Key, ToWorkerType);
+				}
+			}
+		}
+	}
+
+	// Remove any keys for deleted actor groups.
+	TArray<FName> Keys;
+	WorkerAssociation.ActorGroupToWorker.GetKeys(Keys);
+	for (FName Key : Keys)
+	{
+		if (!ActorGroupKeys.Contains(Key))
+		{
+			WorkerAssociation.ActorGroupToWorker.Remove(Key);
+		}
+	}
+
+	FName FirstWorkerType = WorkerTypes->Array()[0];
```

Comment:
Could you define a more sensible default worker type, preferably the unreal server type from GDK constants.